src/axom/sidre/tests/sidre_view.cpp

```diff
@@ -1852,6 +1852,119 @@ TEST(sidre_view, deep_copy_shape)
 
 //------------------------------------------------------------------------------
 
+TEST(sidre_view, reshape_array)
+{
+  using namespace axom;
+  DataStore ds;
+  Group* root = ds.getRoot();
+
+  constexpr int DMAX = 4;
+  IndexType shapeOutput[DMAX];
+  int nDim = -1;
+
+  // A view with array in a buffer.
+  auto* viewA =
+    root->createView("viewA", sidre::detail::SidreTT<int32_t>::id, 12);
+
+  {
+    nDim = viewA->getShape(DMAX, shapeOutput);
+    EXPECT_EQ(viewA->getNumElements(), 12);
+    EXPECT_EQ(nDim, 1);
+    EXPECT_EQ(shapeOutput[0], 12);
+  }
+
+  {
+    // Attempt to reshape empty View should be no-op.
+    EXPECT_TRUE(viewA->isEmpty());
+    IndexType badShape[] = {1, 2, 3};
+    viewA->reshapeArray(3, badShape);
```

Comment:
Note: This is a "no-op", but it will emit a warning. 
Should we add a `SLIC_INFO` that the warning can be ignored?

Reply:
Good idea.  Done.